test/grpc/status_test.exs

```diff
@@ -0,0 +1,30 @@
+defmodule GRPC.StatusTest do
+	use ExUnit.Case
+
+	alias GRPC.Status
+
+  @status_grpc_codes %{
+    ok: 0,
+    cancelled: 1,
+    unknown: 2,
+    invalid_argument: 3,
+    deadline_exceeded: 4,
+    not_found: 5,
+    already_exists: 6,
+    permission_denied: 7,
+    resource_exhausted: 8,
+    failed_precondition: 9,
+    aborted: 10,
+    out_of_range: 11,
+    unimplemented: 12,
+    internal: 13,
+    unavailable: 14,
+    data_loss: 15,
+    unauthenticated: 16
+  }
+
+  test "match an error atom status to a gRPC status defined code" do
+    for {function, expected_value} <- @status_grpc_codes, do:
+      assert expected_value == apply(Status, function, [])
+  end
```

Comment:
might be a good idea to also have a similar test for the status message.

You could list `@status_grpc_codes` (which I'd rename to `@grpc_status_codes`) as a list of tuples `{status_name :: atom, status_code :: integer, message :: String.t()}` so that you can use it in both tests

Reply:
To test the messages status maybe we need to put it on `rpc_error_test.exs` file 'cause all the logic of the messages are in `GRPC.RPCError` module, we can add a test case with this similar approach.

Or if we need to share the same unified list `@grpc_status_codes` on these two files, maybe we need to create this on the factory file to share it between these two test files, what do you think?

Reply:
I'm ok with calling Status.ok and friends in the rpc_error_test.exs file, associating each one with the expected message

Reply:
What do you think about this approach? [4b779a4](https://github.com/elixir-grpc/grpc/pull/314/commits/4b779a4d62fe9a15f014b0007256bc041a70a60b)

Reply:
I took the liberty of changing this so the PR gets merged asap :)
I moved some things around to avoid defining the factory, and also added some more property tests here